mutator/branch/mutateif.go

```diff
@@ -20,17 +22,24 @@ func MutatorIf(pkg *types.Package, info *types.Info, node ast.Node) []mutator.Mu
 	}
 
 	old := n.Body.List
-
-	return []mutator.Mutation{
-		{
-			Change: func() {
-				n.Body.List = []ast.Stmt{
-					astutil.CreateNoopOfStatement(pkg, info, n.Body),
-				}
-			},
-			Reset: func() {
-				n.Body.List = old
+	
+	containsErr := strings.Contains(fmt.Sprintf("%v", n.Cond), "err")
+	containsNilCheck := strings.Contains(fmt.Sprintf("%v", n.Cond), "!= nil")
+	
+	if containsErr && containsNilCheck && len(n.Body.List) == 1 {
+		return nil
+	} else {
+		return []mutator.Mutation{
+			{
+				Change: func() {
+					n.Body.List = []ast.Stmt{
+						astutil.CreateNoopOfStatement(pkg, info, n.Body),
+					}
+				},
+				Reset: func() {
+					n.Body.List = old
+				},
 			},
-		},
+		}
 	}
```

Comment:
No need to put this into an else block! This is because both branches return.

Avoiding else's, where the else returns, helps maintain readability as you add more conditions.

Reply:
Great point, makes sense!